### PR TITLE
Backend for medication concept

### PR DESCRIPTION
**Pull request by jrdegreeff**
Not fully tested. My plan is to have the code reviewed now, and I will fix any bugs while developing the frontend which will be faster than manually testing via postman or something similar.

### CODE REVIEW

File: server/medication/collection.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    return medication;
+  }
+
+  /**
+   * Find a medication by medicationId.
+   *
+   * @param {string} medicationId - The id of the medication record to find
+   * @return {Promise<HydratedDocument<Medication>> | Promise<null>} - The medicationId with the given id, if any
+   */
+  static async findOne(medicationId: Types.ObjectId | string): Promise<HydratedDocument<Medication>> {
```
Comment (isabellapedraza) [inline server/medication/collection.ts:30]: Include null Promise in function's return type

File: server/medication/router.ts [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    res.status(200).json(medicalContacts.map(util.constructMedicationResponse));
+  }
+);
+
+/**
+ * Create a new medication record.
+ *
+ * @name POST /api/medication
+ * 
+ * @param {string} name - The username of the user
```
Comment (isabellapedraza) [inline server/medication/router.ts]: [nit] I believe all the param specifications need to be updated for this function and the following functions

**Review by isabellapedraza — CHANGES_REQUESTED** [review]

File: server/medication/router.ts (hunk 2 above)
Comment (jrdegreeff) [reply server/medication/router.ts]: thanks I'll fix that now. Just FYI this is also the case in all of your routes (I copied from there and it looks like we both missed it until now)

File: server/medication/collection.ts (hunk 1 above)
Comment (jrdegreeff) [reply server/medication/collection.ts:30]: We haven't done this in any of the 7 places where we have this possibility. I'm gonna leave it now for consistency, but if you want to change all of them together at some point feel free.

**Review by isabellapedraza — APPROVED** [review]
lgtm!

**Pull request merged**